File: TableView/iOS-d6ea7e2c8b480117c8fcab1a2d76294942a3e81b/SharedCore/SharedCore/ScheduleTableCell.m
//
//  ScheduleTableCell.m
//  SharedCore
//
//  Created by Jonathan Steele on 5/19/14.
//  Copyright (c) 2014 Jonathan Steele. All rights reserved.
//

#import "ScheduleTableCell.h"
#import "Event.h"

@interface ScheduleTableCell()

@property (nonatomic, weak) IBOutlet UILabel *homeTeam;
@property (nonatomic, weak) IBOutlet UILabel *homeTeamScore;
@property (nonatomic, weak) IBOutlet UILabel *awayTeam;
@property (nonatomic, weak) IBOutlet UILabel *awayTeamScore;
@property (nonatomic, weak) IBOutlet UILabel *date;

@property (nonatomic, readonly) NSDateFormatter *dateFormatter;

@end

@implementation ScheduleTableCell

- (id)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        // Initialization code
    }
    return self;
}

- (void)configureWithEvents:(Event *)event
{
    self.homeTeam.text = event.homeTeam;
    self.homeTeamScore.text = event.homeScore;
    self.awayTeam.text = event.awayTeam;
    self.awayTeamScore.text = event.awayScore;
    self.date.text = [self.dateFormatter stringFromDate:event.date];
}

// On-demand initializer for read-only property.
- (NSDateFormatter *)dateFormatter
{
    static NSDateFormatter *dateFormatter = nil;
    if (dateFormatter == nil) {
        dateFormatter = [[NSDateFormatter alloc] init];
        [dateFormatter setDateFormat:@"MMM dd 'at' hh:mma"];
    }
    return dateFormatter;
}

@end

File: TableView/iOS-d6ea7e2c8b480117c8fcab1a2d76294942a3e81b/Noles Football - Swift/Noles Football/ScheduleTableCell.swift
//
//  ScheduleTableCell.swift
//  Noles Football
//
//  Created by Jonathan Steele on 6/7/14.
//  Copyright (c) 2014 Jonathan Steele. All rights reserved.
//

import UIKit

class ScheduleTableCell: UITableViewCell
{
    @IBOutlet var awayTeam : UILabel
    @IBOutlet var awayScore : UILabel
    @IBOutlet var homeTeam : UILabel
    @IBOutlet var homeScore : UILabel
    @IBOutlet var date : UILabel
    
    init(style: UITableViewCellStyle, reuseIdentifier: String)
    {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        // Initialization code
    }

    override func awakeFromNib()
    {
        super.awakeFromNib()
        // Initialization code
    }
}
